### ISSUE TITLE
Work out combat system

### ISSUE THREAD

**Issue by DaveMS** [DaveMS/MicroRPG#1]
We need to work out how the combat system is going to work.

Have a look at Dominions 5 for some inspiration. The Dominions RNG roll is interesting as it makes anything possible even if it's very unlikely (like one-shotting a dragon).

https://illwiki.com/dom5/combat-mechanics

Ideally we want the player to always feel in danger, even in fights which are "low-level".
We started modelling body parts, but so far these don't do much except allow you to equip items to them.

We should work out what sort of effects injuries to different body parts have.
How are skills going to work which affect combat? At the moment we have only generic attack and defense skills. 

This ticket is all very "wishy-washy" at the moment as there is a lot to think about. 

Document the combat system (and any other systems which will have an effect on combat).

**Comment by TexasHolden792** [comment]
Here are my thoughts:

1.) To be able to decide how this limb-involved combat system works, I think it's important to clarify how healing will work. Presuming that later, a healing class of some kind will be added, what will that actually mean? Can they heal, for example a hand if it's injured? What if the hand is dead?

2.) On that note, for those who are not healing classes, will they have items? I think that at the moment, all the monsters feel quite hard but in a good way in that it feels dangerous. However, without a way to heal myself between fights, it is difficult to sustain. 

3.) Ideas for impediments due to body part loss:

Head loss == death

Body loss == death

Neck loss == death

Finger loss = slightly more chance of missing attack

Hand/Arm loss = only one attack per round (assuming usually two per round) OR   only one attack per two rounds if usually you only get one. Basically, slowing them down. Also consider whether the actual hit does less damage. 

Foot loss = Reduced movement speed. As it is now,  I seem to be able to escape from battles if I decide to move once the monster attacks me. I guess the player will have a movement speed and so will the monsters to determine if they can outrun them? In this case, foot loss could result in movement reductions.

Leg loss = Greatly reduced movement speed and a random chance of bleeding out to death per turn until they can get healing. (Even if healing can't replace limbs, we can assume it would sort of close up the wound, so then they'd still have movement reduction but not random chance of bleeding to death)

Armour can act as a hit point buffer on that body part so that body part doesn't get hit until the 'health' of the armour has depleted to 0

**Comment by DaveMS** [comment]
How it currently works:
https://github.com/DaveMS/MicroRPG/wiki/Combat-Mechanics
